src/m365/aad/commands/m365group/m365group-remove.ts

```diff
@@ -115,6 +109,124 @@ class AadM365GroupRemoveCommand extends GraphCommand {
+    }
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/groups/${id}/drive?$select=webUrl`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const res = await request.get<{ webUrl: string }>(requestOptions);
+    return res.webUrl.substring(0, res.webUrl.lastIndexOf('/'));
+  }
+
+  private async deleteM365GroupSite(logger: Logger, url: string): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Deleting the group site: '${url}'...`);
+    }
+
+    const spoAdminUrl = await spo.getSpoAdminUrl(logger, this.debug);
+    this.spoAdminUrl = spoAdminUrl;
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.spoAdminUrl}/_api/GroupSiteManager/Delete?siteUrl='${url}'`,
+      headers: {
+        'content-type': 'application/json;odata=nometadata',
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    await request.post(requestOptions);
+  }
+
+  private async deleteM365GroupFromRecyclebin(logger: Logger, id: string): Promise<void> {
+    const maxRetries = 10;
+    const intervalInMs: number = 6000;
+
+    for (let retries = 0; retries < maxRetries; retries++) {
+      if (await this.isM365GroupInDeletedItemsList(id)) {
+        await this.removeM365GroupPermanently(logger, id);
+        return;
+      }
+      else {
+        if (this.verbose) {
+          await logger.logToStderr(`Group has not been deleted yet. Waiting and retrying...`);
+        }
+
+        await this.sleep(logger, intervalInMs);
+      }
+    }
+
+    await logger.logToStderr(`Group could not be removed from the recycle bin after all retries.`);
+  }
+
+  private async removeM365GroupPermanently(logger: Logger, id: string): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Group has been deleted and is now available in the deleted items list. Removing permanently...`);
+    }
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/directory/deletedItems/${id}`,
+      headers: {
+        'accept': 'application/json;odata.metadata=none'
+      }
+    };
+
+    await request.delete(requestOptions);
+  }
+
+  private async isM365GroupInDeletedItemsList(id: string): Promise<boolean> {
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/directory/deletedItems/${id}`,
+      headers: {
+        'accept': 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    try {
+      const response = await request.get<{ id: string }>(requestOptions);
+      return Boolean(response && response.id);
+    }
+    catch (error: any) {
+      if (error.response && error.response.status === 404) {
+        return false;
+      }
+      else {
+        throw error;
+      }
+    }
+  }
+
+  private async sleep(logger: Logger, ms: number): Promise<void> {
```

Comment:
This seems like a util function that we should have available centrally to avoid repetition. Let's add it to utils